keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -333,3 +333,202 @@ features makes their future removal much harder.
 ### Release klog 3.0 with removed features
 Removal of those features cannot be done without whole k8s community instead of
 just k8s core components
+
+### Upgrade / Downgrade Strategy
+
+Proposed user impacting changes are as follows:
+* Deprecate and remove subset of klog flags
+* Split log output (previously only stderr) into stdout/stderr based on type (info/error)
```

Comment:
This is a breaking change, why split this? I think it will catch folks with an unpleasant surprise, even if we advertise it.

Reply:
One of our goals for this KEP was proposing good defaults. If we don't make changes to defaults now, it will be much harder to make changes later. 

First I looked through how klog will look like without features we plan to removed I noticed that it will make handling errors and info logs differently impossible. Currently users can send info logs to different output, like send them to separate files. Without the split it will be impossible to that without parsing the logs, which will make the user experience much worse.

Second I found that there is a need to handle error logs different from info logs on the K8s components side. SIG scalability has created a feature request to buffer info logs and only flush on errors. This can be implemented on single log stream, but it would work much better if steams would be separate.

Reply:
Wearing my SIG Scalability hat, I would actually really like to see the info/error logs be splitted. The environment where INFO logs are actually blocking execution of your code is not the place we would like to be at.
FWIW - logs splitted by severity is what we do internally in Google.
So I'm hugely supportive to it.

Now - wearing my prod-readiness hat, the Elana`s argument "breaking-change" is indeed a concerning one. We should really think how to address that.
[Without understanding klog, maybe what we can do is to split them internally, but by-default merge them at the end of the chain, just allow to splitting them if someone wants to do that.]
But we need to have some answer to that concern.

Reply:
So long as we don't make this the new default, I think it's okay to add support for this. But when making a breaking change like this, we can't change this to the default without more data and testing, per our standard graduation process.